mlflow/tracing/fluent.py

```diff
@@ -421,3 +421,79 @@ def f():
     trace_manager = InMemoryTraceManager.get_instance()
     request_id = json.loads(otel_span.attributes.get(SpanAttributeKey.REQUEST_ID))
     return trace_manager.get_span_from_id(request_id, encode_span_id(otel_span.context.span_id))
+
+
+@experimental
+def get_last_active_trace() -> Optional[Trace]:
+    """
+    Get the last active trace in the same process if exists. If there is an active trace,
+    it will return the trace object as **an immutable copy**.
+
+    .. note::
+
+        The last active trace is only stored in-memory for the time defined by the TTL
+        (Time To Live) configuration. By default, the TTL is 1 hour and can be configured
+        using the environment variable ``MLFLOW_TRACE_BUFFER_TTL_SECONDS``.
+
+    .. code-block:: python
+        :test:
+
+        import mlflow
+
+
+        @mlflow.trace
+        def f():
+            pass
+
+
+        f()
+
+        trace = mlflow.get_last_active_trace()
+
+    .. attention::
+
+        This function returns an immutable copy of the original trace that is active or logged
+        in the tracking store. Any changes made to the returned object will not be reflected
+        in the original trace.
+
+        - To mutate the active trace, use :py:func:`mlflow.get_current_active_span` instead.
+        - To modify the already ended trace (while most of the data is immutable after the
+          trace is ended, you can still edit some fields like tags), please use the
+          respective MlflowClient APIs with the request ID of the trace.
+
+        .. code-block:: python
+            :test:
+
+            import mlflow
+
+
+            @mlflow.trace
+            def f():
+                # Use get_current_active_span() to mutate the active trace
+                span = mlflow.get_current_active_span()
+                span.set_attribute("key", "value")
+                pass
+
+
+            f()
+
+            trace = mlflow.get_last_active_trace()
+
+            # Use MlflowClient APIs to mutate the ended trace
+            mlflow.MlflowClient().set_trace_tag(trace.info.request_id, "key", "value")
+
+    Returns:
+        The last active trace if exists, otherwise None.
+    """
+    # If there is an active trace, return it as an immutable trace object.
+    if active_span := get_current_active_span():
```

Comment:
I'm wondering if there're such use cases people wants to export the `active span` as mlflow trace? If not maybe we could just get the last complete trace from TRACE_BUFFER and drop this case (and rename the function to `get_latest_trace`?)

Reply:
I'm not sure if this is super useful or not either, but for consistency with existing `last_active_run()` API: https://github.com/mlflow/mlflow/blob/501856ac57d105dc0b3fbf145746ad8cdd97b205/mlflow/tracking/fluent.py#L535-L548

Reply:
I think last_active_run is the same as get_current_active_span. Can we start without this and add if there're such feature request? Because I feel like `get_last_active_trace` means you're fetching an active trace but it's immutable, it's a bit weird 🤔

Reply:
The `last_active_run` is not same as `get_current_active_span` as it can also fetch the finished run. But, I'm fine with removing the functionality (as we provide an alternative way to do it). I'd still keep the naming for consistency + the `latest` may be a bit ambiguous (e.g. latest accessed, latest status, etc).